Sources/tart/Fetcher.swift

```diff
@@ -2,45 +2,35 @@ import Foundation
 import AsyncAlgorithms
 
 class Fetcher: NSObject, URLSessionTaskDelegate, URLSessionDelegate, URLSessionDataDelegate {
```

Comment:
What do you think of making `Fetcher` fileprivate here and introducing public `URLFetcher` class with static `fetch` method that will take `viaFile` argument.

This way we'll abstract things even more and will hide and limit the complexity of `Fetcher` to this file (e.g. prevent from initializing `Fetcher` somewhere and calling methods twice).

Reply:
See https://github.com/cirruslabs/tart/pull/306/commits/cf180458e371fb69eeec3ba0060d8a2336d428d4.

Reply:
What do you think about the `fileprivate` part and having URLFetcher with only one method?

Reply:
> What do you think about the `fileprivate` part and having URLFetcher with only one method?

I fail to see the advantages of this approach, compared to the solution in https://github.com/cirruslabs/tart/commit/cf180458e371fb69eeec3ba0060d8a2336d428d4. Can you explain?

With https://github.com/cirruslabs/tart/commit/cf180458e371fb69eeec3ba0060d8a2336d428d4, the `Fetcher` also has only one publicly-accessible method and this approach doesn't introduce extraneous classes that might confuse the consumer.

Reply:
`fileprivate` won't be visible to users. Right now one can create a single `Fetcher` instance and try to use it twice.

Reply:
> Right now one can create a single `Fetcher` instance and try to use it twice.

That's actually not an issue any more since there's no shared state stored between `fetch()` invocations.

I've outlined this in https://github.com/cirruslabs/tart/pull/306/commits/2704cbe384f0a0e85f4da5304ede847bc671ac5e by making the method `static`.